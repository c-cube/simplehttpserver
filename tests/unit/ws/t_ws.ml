open Test_util

let read_file file : string =
  let buf = Buffer.create 32 in
  let ic = open_in_bin file in
  Buffer.add_channel buf ic (in_channel_length ic);
  Buffer.contents buf

let apply_masking = Tiny_httpd_ws.Private_.apply_masking

let decode ~key b =
  let buf = Bytes.copy b in
  apply_masking ~mask_key:key buf 0 (Bytes.length buf);
  buf

let () =
  let key = "\x57\x7d\xfd\x95" |> Bytes.unsafe_of_string in
  let content = read_file "masked.data" in
  let decoded = decode ~key (Bytes.unsafe_of_string content) in
  print_endline (Bytes.unsafe_to_string decoded);
  ()
