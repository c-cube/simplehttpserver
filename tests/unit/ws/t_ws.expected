{"keys":["cname:id:$sub.Tactic.@>|.apply/zDzN_HyCqJmbkchXaGIBMGo-zR7PofTRNP_jYOroXmY=","chash:id:TvgBYbVAI8Y5QCh8LXgxTouFoNY7uedPSfb3FbTdaZ4=","chash:id:wSb29KdzBbhaI2MnJZ-vmOu2Y1XLsWw_inGVIKAVkEI=","chash:id:sMODDjirdBa_TYp4wI_0sY5b0GLdDeOr9S8TnCtC0r8=","chash:id:Yg1LFNGPOjYJGBTLniSw0OvEtwCzCa-UT5AhL3YVAqk=","chash:id:oFOoTmXXMsA6sK9aIOJqaRplSK40Tbk8zOgcoyRnMP8=","chash:id:fUMXfZQe3JCnrINUl6sEz17sJLQRSerV-VqXhICZtwo=","chash:id:_T_rY3IpsdpVj-C9avv1lNa2czWUp1T6fX04cbb63Kw=","chash:id:8-67FQrJITG6yP8gifrAkYEBSfSaQ_DCn72ewmSxZvs=","chash:id:f9koocb06JU5G9LAeO1x8lAZxX9QTiusHDsI22rHGP4=","chash:id:CLWYQMuXDJBXZ1hyC35IqQGlaZ1XZuwbvtypiLwXLFk=","chash:id:GwFRKSo8KFBWdQfO98c8hdn4m69UnqrR3fSrv6en_gw=","chash:id:xTv-oQvTDj6VJ470S3xVJ01u7V2hLyWbGs3SntJ_mb8=","chash:id:S6ix_fAr0CnMkVj6iun633RwfnC999mi4Ha8iufFo0Y=","chash:id:LF8zXDkc9tdv-9Akj-HJis6TcBeqkl-OpClGqrTfmzk=","cname:id:Ordinal.+/04UDEnMK3fxAZ9tP96aGY53i9up689NUiEFviKMupYc=","chash:id:VfQ1EZXMET_peidXp520Kzbn8noGw0C2dvqJLGAlB8Y=","chash:id:leOvf_xENwIOWkBRmwdBgpghImZBLh6mYD-cNVu43Ls=","cname:id:Ordinal.is_valid_rec/yYj2TB30Z38LOf0ymx64phy-O_v4u2cgAQSgW539wVE=","cname:id:Ordinal.of_int/rEC61gFwNKb2rlDC6dBtZPwaAHBoLswPLWn4n6phNJQ=","chash:id:AXyqaXiltzQZXD1tNIJ1lVkXwitxZn87o2yoR0Rdhos=","chash:id:Bg6FCptKKopxR0CWW6ELcJLY5QdR89TPgh2LcV9PjPQ=","cname:id:Ordinal.of_list_rec/4aezHWL49QWDDzJEqj1tVngi6ixEPPC852h6Eey--j0=","cname:id:Ordinal.omega/ZCuWQjoygRV2V9Fv5ZJ6rJ-Zw4nuBf9tdowuALtgnfo=","cname:id:Ordinal.omega_omega/riHXh3fX1zg2nrxNImN4So5oirJtSGvQSexyCKVewf4=","cname:id:Ordinal.one/kPQ0ufkZqe9LYB1sTJiU3t-y-rNmrxq3ZAFAGoM3lmU=","cname:id:Ordinal.pair/26J0tJhC8U0qpukIMnWquWzy3rZWrDn58qvziJ-O8zA=","chash:id:f2zkyQ7slDobWkbHHVxyZy0cQs2TzPcEy2j9nULIsoo=","cname:id:Ordinal.quad/qAz3rByIW7_wECcH5PvSXOkcKORqhLnO2wqkBUOERbk=","chash:id:vNMtyCpYP_mrJJD4fokAgbZO4Ts98BXNRArWMc7s5zQ=","cname:id:Ordinal.simple_plus/p-Xe70MN9v5AzhGein9_vjSRGNDj40x4_ZO7LBD8k5A=","chash:id:5VDJgkBcUyZjZCQSbhyxxUvgJOT5W122SRS8we6HjbY=","cname:id:Ordinal.triple/d2YmkjmS2HRXm65w8R2Q4ttNT234BtXdJYBPdSJUF7I=","cname:id:Ordinal.two/S1CZ-WEACIx3qGyk52mA05unnPZjXWA5_fHKmh2OZEw=","cname:id:Ordinal.zero/4dVMLhYEBhh3k2A1KO4G1Kx9c4YhApiEc3RaeDO-WiQ=","cname:id:Ordinal.~$/5LPUnmZCZ-IOGRL_yd8Pf1iHdGLiVt6F6busAj1xvz8=","chash:id:07g7-3b78szJV0_oO_cNo8NjoVIzTdq2vc7bI0A3wfs=","chash:id:_03uTSJGWcJc0lfzYBPNXY74akG0L_TbkZmqgPLQC08=","chash:id:bRL7GkTmmabexPFQQroGW3pi9N9c8n7jgfePux56v_k=","chash:id:uVLERfGJvfioO2-zyvQHhthLg7aNP-s7BTQZk6krQac=","cname:id:Tactic.@>/OQmxCOVkX0iFstCu-76fjzkDFtAo4PN6skEmqzbPqew=","cname:id:Tactic.@>|/ER3pxkbzurAhWNpb30vQpejwoGRg1ExjsnRBgBz77Qw=","chash:id:Cx_02CLNq1iq83kXAOub9LcSZV__bS-mp5Tc60r-lBw=","chash:id:sQVFEJMJ-W7_ufwmhiIIN1VR2XvZnhg34IQPZ6FDwvY=","chash:id:n6IbPr-A9iF_HFR73LLc65wiUj6XLrUuOYnp4JlpTw8=","chash:id:ogXczrII_LIMmjSBPUypvIbG0dJm4cJQFp9cq996xEA=","chash:id:m80u-yYTmE2daleQVFOvEtyp20dKIWvDGigDJV1wZ-0=","chash:id:PDP8cpQzN8TWbzTaTSE6518V3BsfkqtOVs6-OQmeiDY=","cname:id:Tactic.fail/Jjp2CdEZP4CJw_ywvJ4HCojkbltrtRIdR9WzNuwhmxo=","cname:id:Tactic.goal/VRhn1fTAdV9bVbhUnRUOwBRxGnNF9oJj6f6HQFoBoOo=","chash:id:fMtDnfRV0L93IiEBl-iD0ARUPfuVkN6j65zu41KZOc0=","chash:id:_BnJ9g-Hd66BX7CtzYyoiaQp2rrE8Y34YxlIzHt92Go=","chash:id:G7_BbFP4G3JRVnCNUAQvw2WhCU79zNYBYAkoOD774h8=","chash:id:BO4_9DY2Sftp2x_cbO4txgJvHlEtLMOmX-3XhTdYrgU=","cname:id:Tactic.let*/aoEtSUWyx4C1eJGFoTd87irNvV75PhmZDyQR9hrKJMY=","cname:id:Tactic.let+/LFrz0pKnf57tlXx4zP3M7egzoWOyYp8CePVzT2U4hvw=","chash:id:3OYUOkn6oAVMWd524KV3Gy6DFktOKhhquNhkBSrkXCI=","chash:id:4j6W0EW7lfyrzmt-LiWw_8DnQhN0psehMemd2SgpbXc=","chash:id:pC5NGBrCznMq5gujoLW9RzvuF4eG8_J-DABjVfLU2vw=","chash:id:P09oVupzgClnPcO65Bt9t_CBvixl_BSFR1_Pt6mO7Mo=","chash:id:rfFWjHsP2bU5gCV6TLKT3b30qQI7SfOanSnb6WISzk4=","chash:id:M_8R_OyWFjeaV6Yi5XGF5t25SoUDiRHxR0L5KNzsey4=","cname:id:Tactic.return/nOZYJE0c11cVXEZ3dyR3fLmN7CusoXZcWv_E5rj1Sno=","chash:id:HSfpRXgwegI9IzAE0kUw5fsWbDxDEonnfJudB5X9IFU=","chash:id:tGWlUP4W5hlsAlX-LLVWbKYGd_x8_Oh3QkM_Mh0cVkQ=","chash:id:0RDNJKrprwhzZnwv6_4SHdJmjBkUaEl80cRD8KLlobQ=","chash:id:7MjUUbh59AlFOoWq4ydXkVDa2wcxRaHRECXtTiGx97A=","chash:id:UcegWHZs_FohJISYfjgmRStF0wJdmcNipjZNxdbf7kg=","chash:id:1rTDBXbJ9NzmAb1HCzLApbgjnMq9csjb_-T2eo3M9qE=","chash:id:P2NTUh7Iqbu4G9enw2ClZYeCo0sWzeat6LyMVeUb_aM=","chash:id:U9kVyZrCvG5p2paYNJyP00uBpxvNgYfDMkkD5m41oRo=","chash:id:6YtR9zZX2BLhrv_9TibmKJPSxjxIfCOqA-u3j6PgMME=","chash:id:regQAAkL4OiJiA_3WGgwNoekg99GBVlhptISai5FMtc=","cname:id:abs/ovKIgcJdKizSFRE9PrjClN-RfK0U-x94uT9Q3FEOyg0=","chash:id:SJYyfPZN5IQNmgfxvvjdIb-kv03qgqY8_P5wNw-ciTo=","cname:id:compare/cgQsKGxG_YLNpeWFgK_67yRUkcTwObCmJpBdTLNek9o=","chash:id:tlnN-HIHFBl-LahVJgewgdmqx3XMU-nZGYznKui6mbg=","cname:id:count.Ordinal.t/EOyFophGWKkf1hjScha8OaO4aWf1AyRtbLkOrTEJz8Y=","cname:id:count.Peano_nat.t/44mT4uFqWKJ1qCx5Q1cHIBVgYwa4euPzXQ_Q6hJeJyo=","cname:id:count.Tactic.goal/TWQd5QXLrHpktk-8_NRqW6-SmM2bawOPym9y-yXwnX4=","cname:id:count.Tactic.identifier/8Auxsxzg2uabENun1tm8Nx5HNvOmadqYKc8TsYq09Nk=","cname:id:count.Tactic.model/oteg5VB9jWvYJFV3kMfx4v-jNMh1WrlKGiR60jvUtmk=","cname:id:count.Tactic.proof/MPYpLtzqQEmjIsWF6smXmOFto_iSMs6Nbk094gr_XYU=","cname:id:count.Tactic.term/MpmxykFVfzvoHELg9fwQQd5BIhmV3Xti4fW9GVGDfqA=","cname:id:count.bool/DTTOG2K6vxVFmPTr_xCJhE-R4nTMwQh9HS57aQPAtZY=","cname:id:count.float/oNpIj5_DtVMJESxVhmSaDh6HDLEJUGcUJJRGR1DkA8U=","cname:id:count.int/ZNW_6Wy-AUPQ76y1KAyqz45UijSPZqsKgPx30iJb0ME=","cname:id:count.list/e7pWDgXTuULOm_d3Rgjwj_Sxu0_0PCGJXnG6wcUUbDk=","cname:id:count.option/9o7gIEnBNwrxyi82WRvUw-63pOJe_5rdAwbctFHtGAo=","cname:id:count.real/WPWaRiam-ty4VaWTHN4gFT85PrOHSwwr6mW1h53sQ_k=","cname:id:count.result/10U3ym3gHiiFveobmDUCbbXOb1oUxLaV4epuWLjiXkY=","cname:id:count.string/CHvK5EkoypTf1yFkvxAlxQltDh0DuBv2-_2x0aJnb5A=","cname:id:count.unit/w9v31oJaZGKpPhBImU8uqX5Q1rS-nhkjQ58DNmSx78c=","cname:id:explies/VM8_UtUVKJhMRhUdrFKSOKDK3MMGK_3oaqEdwhajfFI=","chash:id:L-dulXdou0hG4hHCKr6AiiCK_9RTLhx_PVF_Ddk4Bv4=","cname:id:i/GSrVa5f2p-J93bYXVODFpv8btZqliNQCOSqUuN99Bis=","chash:id:5_a_5vP826hPuQQ2_YhcB6YIZX6Z9QsnToWYCHCKslU=","chash:id:X5Hr3NZzfXYWrtwtz0zxsNfwEM_ZckO2B0TEHg9MYSo=","chash:id:_hDwsUo8KY2jB7O-qq2KOzwVLExj5HCGEeo7MUPlAQI=","chash:id:qgNIjyPRK1TlDZN8thDzY36PfzkWGSWY99-jjAPE0xk=","chash:id:7Qct4f96pe0sy6HhPGDxvMp-InD7wYc36Ir7aT82wDM=","chash:id:jzgM1-UIoYMtTGe3YEuHLxajU5uM2H0ecyyC3msyHtU=","chash:id:52_A5-4ysJL8u0CTy5b3SNdFg2cPyMVl0I5rtfJ42p4=","chash:id:7lQTa-pUjM0Z3gni9ld0HbVuJC0VZDVQV-jhDx54DTs=","chash:id:QPPrx4rFHU2xIUxKUuTuCQz9uyBjWEh29WYeb-3zUdo=","chash:id:gtxohI9StmYVdoeDwSydOqpu1YOd8c7CEnfv6m0zBPM=","chash:id:Hmkc2zw8I3SG_ewvYhWPYeOo243SbFYR2sKtzqFchBs=","chash:id:nA8cfGZcq_KQTHLlJL4iiRI08LNG34WVeCEw-8TviH8=","chash:id:JzQhBEmFSSUK6z6ItKzKwuw0WrRS9-NGbnWc5MJu5WE=","chash:id:0_uVob2xDiE5pe_CDANbV2d1imtOt1EVBFiVrVlNui0=","chash:id:zDwgMxtiymoZ19hAlS1oPiB3u3_LmecCT_zMke47qQw=","chash:id:rf6ce7GJBHdhq9ZQJO-dZSUkZQERwUUlyD0XNY53jpw=","cname:id:~+/o-a5xt9X9_wdGtTXQB4pzHuPHSdbW8uV2z9LKItKLWU=","chash:id:ZKv5YjyHvMF6Z0xZ2861BpMcdVOlbGIRcxeOsytDiOE="]}

